src/governance/VoltVetoGovernor.sol

```diff
@@ -0,0 +1,398 @@
+        return 2425847; // ~1 year with 1 block every 13s
+    }
+
+    /// @notice State of a given proposal
+    /// The state can be one of:
+    /// - ProposalState.Pending   (0) Lasts only during the block where the veto proposal has been created.
+    /// - ProposalState.Active    (1) If action is pending in the timelock and veto quorum has not been reached yet.
+    /// - ProposalState.Canceled  (2) If a veto was created but the timelock action has been cancelled through another
+    ///   mean before the veto vote succeeded. The internal _cancel() function is not reachable by another mean (no
+    ///   public cancel() function), so this is the only case where a proposal will have Canceled status.
+    /// - ProposalState.Defeated  (3) If proposal already executed or is ready to execute in the timelock.
+    /// - ProposalState.Succeeded (4) If action is pending in the timelock and veto quorum has been reached. Veto can be executed instantly.
+    /// - ProposalState.Executed  (7) If a veto successfully executed.
+    /// note that veto proposals have a quorum that works with 'against' votes, and that only 'against' votes can be
+    /// cast in this veto governor.
+    function state(
+        uint256 proposalId
+    ) public view override returns (ProposalState) {
+        ProposalState status = super.state(proposalId);
+        bytes32 queueid = _timelockIds[proposalId];
+
+        // @dev all proposals that are in this Governor's state should have been created
+        // by the createVeto() method, and therefore should have _timelockIds set, so this
+        // condition check is an invalid state that should never be reached.
+        assert(queueid != bytes32(0));
+
+        // Proposal already executed and stored in state
+        if (status == ProposalState.Executed) {
+            return ProposalState.Executed;
+        }
+        // Proposal cannot be Canceled because there is no public cancel() function.
+        // Vote has just been created, still in waiting period
+        if (status == ProposalState.Pending) {
+            return ProposalState.Pending;
+        }
+
+        // at this stage, status from super can be one of: Active, Succeeded, Defeated
+        // Read timestamp in the timelock to determine the state of the proposal
+        uint256 timelockOperationTimestamp = _timelock.getTimestamp(queueid);
+
+        // proposal already cleared from the timelock by something else
+        if (timelockOperationTimestamp == 0) {
+            return ProposalState.Canceled;
+        }
+        // proposal already executed in the timelock
+        if (timelockOperationTimestamp == 1) {
+            return ProposalState.Defeated;
+        }
+
+        // proposal still in waiting period in the timelock
+        if (timelockOperationTimestamp > block.timestamp) {
+            // ready to veto
+            if (_quorumReached(proposalId) && _voteSucceeded(proposalId)) {
```

Comment:
remove unnecessary `_voteSucceeded` check as `_quorumReached` is all the verification needed